mage_integrations/mage_integrations/sources/api/__init__.py

```diff
@@ -199,6 +199,12 @@ def load_data(
             df = self._deal_with_google_sheets(response, separator, header)
             yield df.to_dict(orient='records')
 
+        elif checked_type == 'application/gzip':
+            df = polars.read_csv(BytesIO(response.content),
```

Comment:
Don't you need to decompress the data first?

Reply:
It seems polars decompress .gz files by default
Rust crate link: https://docs.rs/polars/latest/polars/